doc/gen_support_matrix_html.rs

```diff
@@ -0,0 +1,426 @@
+#!/usr/bin/env -S cargo -Zscript
+---cargo
+[package]
+edition = "2021"
+
+[dependencies]
+argh = { version = "0.1.12" }
+miette = { version = "7.2.0", features = ["fancy"] }
+minijinja = { version = "2.0.3" }
+serde = { version = "1.0.0", features = ["derive"] }
+serde_yaml = { version = "0.9.34" } # TODO: use a maintained crate instead
+thiserror = { version = "1.0.61" }
+---
+
+use std::{fs, io, path::{Path, PathBuf}};
+
+use serde::Serialize;
+use miette::Diagnostic;
+
+const TABLE_TEMPLATE: &str =
+r##"<!-- This table is auto-generated. Do not edit manually. -->
+<table>
+  <thead>
+    <tr>
+      <th>Chip</th>
+      <th>Testing Board</th>
+      <th colspan="{{ matrix.functionalities|length }}">Functionality</th>
+    </tr>
+    <tr>
+      <th></th>
+      <th></th>
+      {%- for functionality in matrix.functionalities %}
+      <th>{{ functionality.title }}</th>
+      {%- endfor %}
+    </tr>
+  </thead>
+  <tbody>
+    {%- for board in boards %}
+    <tr>
+      <td>{{ board.chip }}</td>
+      <td>{{ board.name }}</td>
+      {%- for functionality in board.functionalities %}
+      <td class="support-cell">{{ functionality.icon }}</td>
```

Comment:
```suggestion
      <td class="support-cell" title="{{ functionality.description }}">{{ functionality.icon }}</td>
```

if that information is available there